### PR TITLE
Razer Mamba Elite initial support

### PR DESCRIPTION
**Pull request by IvanVojtko**
This PR is to enable Mamba Elite in openrazer. At first you need to enable driver mode:
`echo -n -e "\x00\x02" > device_mode`
and then you are ready to play with your device. You can set color:
`echo -n -e "\xff\x00\x00" > matrix_effect_static`
or you can set brightness
`echo -n "100" > matrix_brightness`
Also if you lock screen, all lights will be turned off. After unlocking, all light will resume. It's more like a work in progress, I'm sure it needs som cleanup and maybe rethink few things, but I think that it's good as a starting point. Other advanced effects needs to be client driven. Client needs to constantly spam your device with colors and brightness and more. I can probably do this, but it will be a huge change in code.

### CODE REVIEW

File: driver/razercommon.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ do { \
 #define RED_PROFILE_LED   0x0C
 #define GREEN_PROFILE_LED 0x0D
 #define BLUE_PROFILE_LED  0x0E
+#define MAMBA_ELITE       0x0F
```
Comment (z3ntu) [inline driver/razercommon.h:57]: 0x0f is the `Docked LED`

**Review by z3ntu — CHANGES_REQUESTED** [review]
Yeah it's pretty hacky as you said. I have a few other (minor) comments that I didn't write now as the daemon needs "spamming support" for this to be merged :)

**Comment by IvanVojtko** [comment]
"Spamming support" is not a big deal. I am more worried about generating colors. Maybe we should create new thread, that is waiting for commands and spamming colors to devices? If so, I can take a look.

File: driver/razercommon.h (hunk 1 above)
Comment (IvanVojtko) [reply driver/razercommon.h:57]: Yep. This device doesn't have any LED ID. I just set it to some random value to distinguish it during debugging. Will rewrite this part.

**Comment by z3ntu** [comment]
That sounds like what I'm doing in my [razer_test](https://github.com/z3ntu/razer_test) project even though I suspend the thread when it's not used.
"Generating colors" / re-implementing effects is the harder part :) I've only done [Spectrum and Wave for now in razer_test](https://github.com/z3ntu/razer_test/tree/master/src/customeffect)

You can probably also get inspiration at the uchroma project (though: don't copy-paste code as I don't want to add 500 extra python dependencies :wink: )

**Comment by Donelle** [comment]
I have an Elite and would like to have this device added to the list

**Comment by tildearrow** [comment]
Isn't it possible to add support to the device in the DKMS driver, but only with "matrix_custom_frame" and "matrix_effect_custom"?

**Comment by sandrwich** [comment]
When can we expect this to be fully ready and merged?

**Comment by beranat** [comment]
Dear *[Ivan Vojtko](https://github.com/IvanVojtko)*, could you check the patch [with more effects and left/right colored lines colors](https://github.com/openrazer/openrazer/pull/960).

**Comment by IvanVojtko** [comment]
Hi @beranat I'@e checked it and tried it and it looks perfect! Good work! I only had to "hack fix" polychromatic, but this is another project (Python 3.8 removed some methods...). For me it's perfect. We need to wait for @z3ntu, what he is thinking. I've tried reactive, breathing, wave, static and everything was working perfect.

**Comment by beranat** [comment]
Dear **Ivan**, have you face problem with [patched polychromatic from my repo](https://github.com/beranat/polychromatic)?

**Comment by z3ntu** [comment]
So this is superseded by #960 ? If yes, please close this PR.

**Pull request closed without merging**